client/src/components/MainAppBar/MainAppBar.tsx

```diff
@@ -1,20 +1,27 @@
 import React, {useEffect} from "react";
-import {useAppDispatch} from "../store/hooks";
-import {logoutThunk} from "../store/auth/auth.slice";
-import {ImageWrapper, StyledAppBar, StyledIcon, StyledImage, Wrapper} from "../styles/MainAppBarStyles";
-import {upsertFavoritesThunk} from "../store/favorites/favorites.slice";
-import {getFavorites} from "../services/favorites.service";
-import {Item} from "../models/item";
-
-const logo = require('../../assets/logo-spic.png');
+import {useAppDispatch} from "../../store/hooks";
+import {logoutThunk} from "../../store/auth/auth.slice";
+import {ImageWrapper, StyledAppBar, StyledIcon, StyledImage, Wrapper} from "./MainAppBarStyles";
+import {Item} from "../../models/item";
+import {getFavorites} from "../../services/favorites.service";
+import {upsertFavoritesThunk} from "../../store/favorites/favorites.slice";
+import {getUserDetails, UserDetails} from "../../services/user-settiings.service";
+import {updateUserDetailsThunk} from "../../store/user-details/user-details.slice";
+
+const logo = require('../../../assets/logo-spic.png');
 export const MainAppBar = () => {
     const dispatch = useAppDispatch();
 
     useEffect(() => {
         const fetchData = async () => {
             const favorites: Item[] = await getFavorites();
-            const filteredFavorites: string[] = favorites.map((item: { _id: string; }) => item._id);
+            const filteredFavorites: string[] = favorites?.map((item: { _id: string; }) => item._id);
+
+            const userDetails: UserDetails = await getUserDetails();
```

Comment:
We should make precautions over the application whenever we try to access `userDetails` fields before actually getting this response (for example - בוקר טוב <שם>, מה <תרצה/תרצי> לעשות היום?)

Reply:
Not understand the comment